### PR TITLE
feat(DataTestDirective): using hqrDataTest in HTML templates sets data-test attribute

### PR DESCRIPTION
**Pull request by hitjim**
We're adopting `data-test` as the new "test automation ID" attribute. This lets us...

1. Take a more deliberate approach to making these useful
2. Know where we've reviewed and refactored test IDs (i.e. `data-auto-id` may still be useful, but it likely won't be recently reviewed by test and dev engineers for validity)
3. Take a staged approach to deprecating `data-auto-id`, since it's frequently used in a fragile or noisy way.

This is ready to go for final review. It adds a simple directive that will take something like this in your template...
```html
<div hqrDataTest="first-test-id">My high school poetry</div>
<app-table hqrDataTest="how-did-you-get-this">
  (... eye-roll fuel...)
</app-table>
```

into ...
```html
<div data-test="first-test-id">My high school poetry</div>
<app-table hqrDataTest="how-did-you-get-this">
  (... eye-roll fuel...)
</app-table>
```

... without needing to add `@Input() hqrDataTest: string;` to every single component in this library, or set `[attr.data-test]` in other apps.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@bellese/angular-design-system",
-  "version": "5.0.0",
+  "version": "5.0.2",
```
Comment (hitjim) [inline package-lock.json:3]: This is OK, right?

I forget if this is what blows up versions in the GH release bot, or if it's when we change the version manually in `package.json`

File: src/app/modules/alert/alert.component.ts [hunk 2]
```diff
@@ -11,7 +11,22 @@ export class AlertComponent {
...
 
+    ngOnInit() {
+      if (this.dataTest) {
+        this.dataTestVals = {
+          outer: `${this.dataTest}-${this.variation}`,
+          close: `${this.dataTest}-close`,
+          heading: `${this.dataTest}-heading`,
+        };
+      }
+    }
```
Comment (dianadean) [inline src/app/modules/alert/alert.component.ts:31]: This seems like a lot overhead for developers when the testers can just as easily get the right selector without multiple test ids.  If we went the route of directives...I think a lot of this can go away.  So say we have this alert component, the test Id could simply be on the component itself:

```
<app-alert testId='profile-page-alert'>
...
<app-alert>
```

The testers could then do something like (this may not be the exact syntax):

```
 cy.get(app-alert[testId='profile-page-alert'])
  .find('button') 
  .click()
```

This isn't brittle because we're already looking in the context of app-alert and there's only one button so it just seems like overkill to add another test id.
Say it was a more complex component with multiple buttons, we could then select say the close button:
`.find('button.ds-c-dialog__close')`. In the case of the header, they could use `.find('.ds-c-alert__heading')`.

Again I don't think these are brittle selectors because I doubt the design system is ever going to change these classes and they're not coupled to a style.

So basically, I'm wondering if we can use a happy medium where we use non-brittle (class) selectors where they make sense but use the test ids to make it easier to get to the actual component that's being tested.

If none of this makes sense, let me know and we can huddle and talk through it.

**Review by dianadean — COMMENTED** [review]
Left a couple of comments.  Reach out if you want to talk through them.

File: package-lock.json (hunk 1 above)
Comment (hitjim) [reply package-lock.json:3]: @mcelestini ? ⬆️ re: when / where to bump version numbers or not?

File: src/app/modules/alert/alert.component.ts [hunk 3]
```diff
@@ -11,7 +11,22 @@ export class AlertComponent {
     @Input() hideIcon = false;
     @Input() hideClose = false;
     @Input() dataAutoId: string;
+    @Input() dataTest: string;
```
[the remark this answers is not in the document]
Comment (hitjim) [reply src/app/modules/alert/alert.component.ts]: @dianadean took a stab at this... LMK

**Review by dianadean — APPROVED** [review]

File: src/app/modules/alert/alert.component.ts (hunk 2 above)
Comment (hitjim) [reply src/app/modules/alert/alert.component.ts]: Another thought since our chat, and while cleaning up to go this direction ... suggesting usage of inner `class` or other existing attributes to find internal elements of `shared UI` or even `app` components will encourage better practices anyhow.

We'll probably want to provide this angle as motivation to create specific "wrapper" CSS classes that describe their purpose. We do that better in this repo than we do in other apps, and it would be a good habit to spread, not just for test automation.

**Review by dianadean — APPROVED** [review]
